plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/bootstrap/tls/DefaultSslContextProvider.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.arrow.flight.bootstrap.tls;
+
+import org.opensearch.plugins.SecureTransportSettingsProvider;
+
+import javax.net.ssl.SSLException;
+
+import java.util.Locale;
+
+import io.grpc.netty.shaded.io.netty.handler.ssl.ApplicationProtocolConfig;
+import io.grpc.netty.shaded.io.netty.handler.ssl.ApplicationProtocolNames;
+import io.grpc.netty.shaded.io.netty.handler.ssl.ClientAuth;
+import io.grpc.netty.shaded.io.netty.handler.ssl.SslContext;
+import io.grpc.netty.shaded.io.netty.handler.ssl.SslProvider;
+import io.grpc.netty.shaded.io.netty.handler.ssl.SupportedCipherSuiteFilter;
+
+/**
+ * DefaultSslContextProvider is an implementation of the SslContextProvider interface that provides SSL contexts based on the provided SecureTransportSettingsProvider.
+ */
+public class DefaultSslContextProvider implements SslContextProvider {
+
+    private final SecureTransportSettingsProvider secureTransportSettingsProvider;
+
+    /**
+     * Constructor for DefaultSslContextProvider.
+     * @param secureTransportSettingsProvider The SecureTransportSettingsProvider instance.
+     */
+    public DefaultSslContextProvider(SecureTransportSettingsProvider secureTransportSettingsProvider) {
+        this.secureTransportSettingsProvider = secureTransportSettingsProvider;
+    }
+
+    /**
+     * Returns true to indicate that SSL is enabled.
+     * @return true
+     */
+    @Override
+    public boolean isSslEnabled() {
+        return true;
+    }
+
+    // TODO - handle certificates reload
```

Comment:
Sorry to jump into this PR with my comment, but I noticed your todo and I might be able to help on this area. I have built a library which can take care of reloading the sslcontext and has also an adapter for Netty. Please have a look here if you are interested:
- https://github.com/Hakky54/sslcontext-kickstart?tab=readme-ov-file#support-for-reloading-ssl-at-runtime
- https://github.com/Hakky54/sslcontext-kickstart?tab=readme-ov-file#additional-mappers-for-specific-libraries

Reply:
That's great. I will take a look. Thank you @Hakky54